test/integration/dockerregistryclient_test.go

```diff
@@ -3,12 +3,54 @@
 package integration
 
 import (
+	"fmt"
+	"net"
+	"net/http"
+	"net/url"
 	"reflect"
 	"testing"
+	"time"
 
 	"github.com/openshift/origin/pkg/dockerregistry"
 )
 
+// isRegistryReachable returns false if a Get on registry url fails with similar error:
+//   Get https://registry.com/v2/: dial tcp registry.com:443: i/o timeout
+//   Get https://registry.com/v2/: dial tcp: lookup registry.com: no such host
+//   Get https://registry.com/v2/: dial tcp registry.com:443: getsockopt: connection refused
+//   Get https://registry.com/v2/: read tcp 127.0.0.1:39849->registry.com:443: read: connection reset by peer
+func isRegistryReachable(registry, version string, allowInsecure bool) (bool, string) {
+	client := http.Client{Timeout: time.Second * 4}
+	_, err := client.Get(fmt.Sprintf("https://%s/%s/", registry, version))
+	if err != nil && allowInsecure {
+		_, err = http.Get(fmt.Sprintf("http://%s/%s/", registry, version))
+	}
+	errStr := ""
+	if err != nil {
+		switch t := err.(type) {
+		case *url.Error:
+			switch t := t.Err.(type) {
+			case *net.OpError:
+				if t.Op == "dial" || t.Op == "read" {
+					return false, err.Error()
+				}
+			}
+		}
+		errStr = err.Error()
+	}
+	return true, errStr
+}
+
+// requireReachableRegistry allows to skip a test if given registry cannot be reached or resolved.
+func requireReachableRegistry(t *testing.T, registry, version string, allowInsecure bool) {
+	reachable, err := isRegistryReachable(registry, version, allowInsecure)
+	if !reachable {
+		t.Skip("skipping due to unreachable registry %s: %v", registry, err)
```

Comment:
`hack/test-integration.sh` shows just `ok`:

    $ printf "127.0.0.2 index.docker.io docker.io" >> /etc/hosts
    $ ./hack/test-integration.sh '^TestRegistry'

    Test test/integration -tags='integration !docker etcd' ...

    Building test executable...
    Starting etcd...
    etcd: {"etcdserver":"2.1.1","etcdcluster":"2.1.0"}
    {"message":"Internal Server Error"}
    ok      TestRegistryClientConnect
    ok      TestRegistryClientConnectPulpRegistry
    ok      TestRegistryClientDockerHubV2
    ...

I can see the skip message only when I run the tests manually:

    $ go test -test.run TestRegistryClient -v ./test/integration/dockerregistryclient_test.go 
    === RUN TestRegistryClientConnect
    --- PASS: TestRegistryClientConnect (0.00s)
    === RUN TestRegistryClientConnectPulpRegistry
    --- PASS: TestRegistryClientConnectPulpRegistry (3.71s)
    === RUN TestRegistryClientDockerHubV2
    --- SKIP: TestRegistryClientDockerHubV2 (0.00s)
            dockerregistryclient_test.go:48: skipping due to unreachable registry index.docker.io: Get https://index.docker.io/v2/: dial tcp 127.0.0.2:443: connection refused
    === RUN TestRegistryClientDockerHubV1
    ...

I'll shorten the skip message a bit.